sdks/java/extensions/avro/build.gradle

```diff
@@ -63,67 +74,91 @@ dependencies {
   testImplementation library.java.junit
   testRuntimeOnly project(path: ":runners:direct-java", configuration: "shadow")
   testRuntimeOnly library.java.slf4j_jdk14
-  avroVersions.each {
-    "avroVersion$it.key" "org.apache.avro:avro:$it.value"
-    "avroVersion$it.key" "org.apache.avro:avro-tools:$it.value"
+  avroVersions.each { k,v ->
+    "avroVersion$k"(project(path: ":sdks:java:core", configuration: "shadowTest")) {
+      // Exclude Avro dependencies from "core" since Avro support moved to this extension
+      exclude group: "org.apache.avro", module: "avro"
+    }
+    "avroVersion$k" library.java.junit
+    "avroVersion$k" project(path: ":runners:direct-java", configuration: "shadow")
+    "avroVersion$k" library.java.slf4j_jdk14
+    "avroVersion$k" "org.tukaani:xz:1.9" // makes as provided since 1.9
+    "avroVersion$k" library.java.zstd_jni // makes as provided since 1.9
```

Comment:
Avro version `1.9` and above mark `xz` as optional. We can't run the compression tests otherwise.
Also include `zstd` and run tests for avro `1.9` and above